include/CL/opencl.hpp

```diff
@@ -1364,15 +1375,15 @@ inline cl_int getInfoHelper(Func f, cl_uint name, T* param, int, typename T::cl_
 #define CL_HPP_PARAM_NAME_INFO_2_1_(F) \
     F(cl_platform_info, CL_PLATFORM_HOST_TIMER_RESOLUTION, size_type) \
     F(cl_program_info, CL_PROGRAM_IL, cl::vector<unsigned char>) \
```

Comment:
Not necessarily a problem, but I'm curious about the choice of cl::vector<unsigned char> vs string. Most of the queries use string where char[] is specified, but I see the following and maybe a couple others use vectors:
CL_PROGRAM_IL_KHR
CL_PROGRAM_IL
CL_PROGRAM_BINARY_SIZES

Reply:
It's because these queries return binary data that isn't necessarily guaranteed to be null-terminated.

Interestingly the spec for [CL_PROGRAM_IL](https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_PROGRAM_IL) says the return value is a `char[]`, whereas the query for CL_PROGRAM_IL_KHR returns an `unsigned char[]`.  In practice, I'm not sure that it matters, but the difference is a little odd, and these C++ bindings return a `cl::vector<unsigned char>` in both cases for whatever that's worth.